libs/dyn/visit.go

```diff
@@ -98,7 +126,7 @@ func (component pathComponent) visit(v Value, prefix Path, suffix Pattern, opts
 		case KindNil:
 			return InvalidValue, cannotTraverseNilError{path}
 		default:
-			return InvalidValue, fmt.Errorf("expected a map to index %q, found %s", path, v.Kind())
+			return InvalidValue, expectedMapAtIndexError{p: path, v: v}
```

Comment:
Nuance: the error is "expected a map **to** index [key]" but the type name is different.